packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx

```diff
@@ -53,15 +49,36 @@ const AccountRecoveryConfirmKey = (_: RouteComponentProps) => {
   // so we set its value after the first request for subsequent requests.
   const [fetchedResetToken, setFetchedResetToken] = useState('');
   const [isFocused, setIsFocused] = useState(false);
+  // For initial page load
+  const [showLoadingSpinner, setShowLoadingSpinner] = useState(true);
   // We use this to debounce the submit button
   const [isLoading, setIsLoading] = useState(false);
   const account = useAccount();
   const ftlMsgResolver = useFtlMsgResolver();
-  const { linkStatus, setLinkStatus, requiredParams } =
-    useAccountRecoveryConfirmKeyLinkStatus();
   const location = useLocation();
   const navigate = useNavigate();
 
+  useEffect(() => {
+    const checkPasswordForgotToken = async (token: string) => {
+      try {
+        const isValid = await account.resetPasswordStatus(token);
+        if (isValid) {
+          setLinkStatus(LinkStatus.valid);
+
+          setShowLoadingSpinner(false);
+          logPageViewEvent(viewName, REACT_ENTRYPOINT);
+        } else {
+          setLinkStatus(LinkStatus.expired);
+        }
+      } catch (e) {
+        console.log(e);
```

Comment:
Missed console log?

Also why not also hide the loading spinner when the link is expired or damaged?

Reply:
If we hid the loading spinner before the other component renders, we could get a flash of the `AccountRecoveryConfirmKey` page  before the `LinkExpired` component loads.